main.go

```diff
@@ -229,12 +235,52 @@ func visit(url *url.URL) {
 				// 30x but no Location to follow, give up.
 				return
 			}
-			log.Fatal("unable to follow redirect: %v", err)
+			log.Fatalf("unable to follow redirect: %v", err)
 		}
 		visit(loc)
 	}
 }
 
+func responseBody(response io.ReadCloser) string {
```

Comment:
this probably should just be a reader, not a read closer, the function should do one thing which is read the body, not two; read and close. That can be done in a wrapper.